cmd/internal/opentelemetry/wrapperMux.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opentelemetry
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/google/cadvisor/version"
+	"os"
+
+	"go.opentelemetry.io/otel/api/correlation"
+	"go.opentelemetry.io/otel/api/global"
+	"go.opentelemetry.io/otel/api/kv"
+	"go.opentelemetry.io/otel/api/propagation"
+	apiTrace "go.opentelemetry.io/otel/api/trace"
+	"go.opentelemetry.io/otel/exporters/otlp"
+	"go.opentelemetry.io/otel/plugin/othttp"
+	"go.opentelemetry.io/otel/sdk/trace"
+	"google.golang.org/grpc/credentials"
+	"k8s.io/klog/v2"
+	"net/http"
+	"strconv"
+	"strings"
+)
+
+const traceServiceName = "cadvisor"
+
+var traceProbability = flag.Float64("trace_probability", 1, "used for sampler (0=never, 1=always and probability is 0.1 to 0.99)")
+
+func InitTrace(traceEndpoint, collectorCert string) *otlp.Exporter {
+	klog.Infof("write telemetry to %s", traceEndpoint)
+	exporterOptions := []otlp.ExporterOption{otlp.WithAddress(traceEndpoint)}
+
+	if collectorCert != "" {
+		creds, err := credentials.NewClientTLSFromFile(collectorCert, "")
+		if err != nil {
+			klog.Fatal(err)
+		}
+		exporterOptions = append(exporterOptions, otlp.WithTLSCredentials(creds))
+	} else {
+		exporterOptions = append(exporterOptions, otlp.WithInsecure())
+	}
+
+	exporter, err := otlp.NewExporter(exporterOptions...)
+	if err != nil {
+		klog.Fatal(err)
+	}
+
+	sampler := getTraceSamplerProbability() //*traceSampler, *traceSamplerParam)
+	providerOpts := []trace.ProviderOption{
+		trace.WithBatcher(exporter),
+		trace.WithConfig(trace.Config{DefaultSampler: sampler}),
+	}
+
+	traceProvider, err := trace.NewProvider(providerOpts...)
+	if err != nil {
+		klog.Fatal(err)
+	}
+	global.SetTraceProvider(traceProvider)
+
+	apiB3Single, apiB3, traceCtx, corrCtx := apiTrace.B3{SingleHeader: false}, apiTrace.B3{SingleHeader: false}, apiTrace.DefaultHTTPPropagator(), correlation.DefaultHTTPPropagator()
```

Comment:
How is `apiB3Single` different from `apiB3`? They seem to be identically configured instance of the same struct.